lib/ui/window.dart

```diff
@@ -286,31 +543,212 @@ class Locale {
     }
   }
 
+  // Unicode language variant codes.
+  //
+  // This iterable provides variants normalized to lowercase and in sorted
+  // order, as per the Unicode LDML specification.
+  //
+  // FIXME: decide whether this returns a single string or something fancier.
+  Iterable<String> get variants => _variants ?? const <String>[];
+
+  // The private _variants field must have variants in lowercase and already
+  // sorted: constructors must construct it as such.
+  final List<String> _variants;
+
+  // A map representing all Locale Identifier extensions.
+  //
+  // Keys in this ordered map must be sorted, and both keys and values must all
+  // be lowercase: constructors must construct it as such.
+  //
+  // This map simultaneously reprents T-extensions, U-extensions, other
+  // extensions and the private use extensions. Implementation detailsf:
+  //
+  // * The 't' entry represents the optional "tlang" identifier of the T
+  //   Extension. If the T Extension is present but has no tlang value, the 't'
+  //   map entry's value must be an empty string.
+  // * The 'u' entry represents the optional attributes of the U Extension. They
+  //   must be sorted in alphabetical order, separated by hyphens, and be
+  //   lowercase. If the U Extension is present but has no attributes, the 'u'
+  //   map entry's value must be an empty string.
+  // * U-Extension keyword keys, matching _reKey, and T-Extension fields in the
+  //   map, whos field separator subtags match _reTkey, are directly entered
+  //   into this map. (These regular expressions don't match the same keys.)
+  // * Other singletons are entered directly into the map, with all
+  //   values/attributes associated with that singleton as the map entry's
+  //   value.
+  final collection.LinkedHashMap<String, String> _extensions;
```

Comment:
shouldn't extensions be exposed?

Reply:
Yes. Eventually - but I haven't yet made a proposal for the correct API.

How do you feel about the balance between several-incremental-steps vs one-big-change? (The former seeks to ease reviews, but could hide the bigger picture one is working towards.)